### DISCUSSION TITLE
Vaidating TLS certificate of webserver using a locally stored one

### DISCUSSION THREAD

**Discussion by ESPnerd** [micropython/micropython#12054] [ESP8266]
I've gotten my board to connect over HTTPS no problem (posting data to the webserver via urequests.post() ) but it would be nice to be able to validate the TLS certificate.  I've found some previous conversations about going about it, but not finding much in the way of examples for an easy implementation.

**Comment by bidrohini** [comment]
Hello, which board are you using particularly? I mean which version of the ESP8266?

**Comment by ESPnerd** [comment]
A Wemos D1 board using an ESP12-F.

**Comment by robertfeldt** [comment]
Can you provide any code/details on how you've gotten HTTPS posts done on a D1 mini board? I have tried many approaches on a D1 mini clone from makershop.de without success so example code would be helpful, to check I haven't missed something. I typically get the OSError 'CONN_LOST' when calling wrap_socket.
